src/lapack/backends/cusolver/cusolver_batch.cpp

```diff
@@ -937,77 +1690,106 @@ std::int64_t getri_batch_scratchpad_size<std::complex<double>>(sycl::queue &queu
+    }
+
+GETRS_GROUP_LAUNCHER_SCRATCH(float)
+GETRS_GROUP_LAUNCHER_SCRATCH(double)
+GETRS_GROUP_LAUNCHER_SCRATCH(std::complex<float>)
+GETRS_GROUP_LAUNCHER_SCRATCH(std::complex<double>)
+
+#undef GETRS_GROUP_LAUNCHER_SCRATCH
+
+template <typename Func>
+inline void geqrf_batch_scratchpad_size(const char *func_name, Func func, sycl::queue &queue,
+                                        std::int64_t *m, std::int64_t *n, std::int64_t *lda,
+                                        std::int64_t group_count, std::int64_t *group_sizes,
+                                        int *scratch_size) {
+    auto e = queue.submit([&](sycl::handler &cgh) {
+        onemkl_cusolver_host_task(cgh, queue, [=](CusolverScopedContextHandler &sc) {
+            auto handle = sc.get_handle(queue);
+            int group_scratch_size = 0;
+            *scratch_size = 0;
+            cusolverStatus_t err;
+
+            // Get the maximum scratch_size across the groups
+            for (int64_t group_id = 0; group_id < group_count; ++group_id) {
+                CUSOLVER_ERROR_FUNC_T(func_name, func, err, handle, m[group_id], n[group_id],
+                                      nullptr, lda[group_id], &group_scratch_size);
+                *scratch_size =
+                    group_scratch_size > *scratch_size ? group_scratch_size : *scratch_size;
+            }
+        });
+    });
+    e.wait();
 }
-template <>
-std::int64_t orgqr_batch_scratchpad_size<double>(sycl::queue &queue, std::int64_t *m,
-                                                 std::int64_t *n, std::int64_t *k,
-                                                 std::int64_t *lda, std::int64_t group_count,
-                                                 std::int64_t *group_sizes) {
-    throw unimplemented("lapack", "orgqr_batch_scratchpad_size");
+
+#define GEQRF_GROUP_LAUNCHER_SCRATCH(TYPE, CUSOLVER_ROUTINE)                               \
+    template <>                                                                            \
+    std::int64_t geqrf_batch_scratchpad_size<TYPE>(                                        \
+        sycl::queue & queue, std::int64_t * m, std::int64_t * n, std::int64_t * lda,       \
+        std::int64_t group_count, std::int64_t * group_sizes) {                            \
+        int scratch_size;                                                                  \
+        geqrf_batch_scratchpad_size(#CUSOLVER_ROUTINE, CUSOLVER_ROUTINE, queue, m, n, lda, \
+                                    group_count, group_sizes, &scratch_size);              \
+        return scratch_size;                                                               \
+    }
+
+GEQRF_GROUP_LAUNCHER_SCRATCH(float, cusolverDnSgetrf_bufferSize)
+GEQRF_GROUP_LAUNCHER_SCRATCH(double, cusolverDnDgetrf_bufferSize)
+GEQRF_GROUP_LAUNCHER_SCRATCH(std::complex<float>, cusolverDnCgetrf_bufferSize)
+GEQRF_GROUP_LAUNCHER_SCRATCH(std::complex<double>, cusolverDnZgetrf_bufferSize)
+
```

Comment:
getrf -> geqrf

Reply:
Thank you for catching the mistake! I have fixed the scratch function

Reply:
Is `GetrsBatchStride` still failing? I found a mistake where there was a call to free in the queue submit and not the host_task, this was fixed in previous commit.